lib/volcanic/authenticator/v1/subject.rb

```diff
@@ -23,15 +23,27 @@ def privileges_for(subject, service)
             exception,
             "#{path}/privileges?filter[subject]=#{subject}&filter[service]=#{service}"
           )
-          return [] if response.blank?
+          privilege_parser(response)
+        end
+
+        # similar to +privileges_for+, accept it
+        # search privileges by token scope
+        def privileges_for_scope(token_base64, service)
+          response = perform_get_and_parse(exception, "#{path}/privileges/identity&filter[service]=#{service}", token_base64)
+          privilege_parser(response)
+        end
+
+        private
+
+        def privilege_parser(res)
```

Comment:
This code isn't about subjects... it is about privileges...

It should probably be in the `Privilege` class